### ISSUE TITLE
No benchmarks are executed by default

### ISSUE THREAD

**Issue by vyazelenko** [melix/jmh-gradle-plugin#6]
By default **jmh** task will not find any benchmarks to execute. 

For example here is [gist] (https://gist.github.com/vyazelenko/5ea163a1b14bc8ea41f5) that contains `gradle.build` and `SampleBenchmark.java`. If I try to execute `gradle jmh` I get the following output:
```cmd
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:compileJmhJavawarning: Supported source version 'RELEASE_6' from annotation processor 'org.openjdk.jmh.generators.BenchmarkProcessor' less than -source '1.7'
1 warning

:processJmhResources UP-TO-DATE
:jmhClasses
:jmhJar
:jmhNo matching benchmarks. Miss-spelled regexp?
Use EXTRA verbose mode to debug the pattern matching.


BUILD SUCCESSFUL

Total time: 8.193 secs
```

However when I do `gradle jmhJar` followed by command line invocation (i.e. `java -jar build\libs\jmh-test-jmh.jar`) then execution is started:
```cmd
P:\Projects\OpenSource\jmh-test>java -jar build\libs\jmh-test-jmh.jar
# VM invoker: C:\Program Files\Java\jre7\bin\java.exe
# VM options: <none>
# Warmup: 5 iterations, 1 s each
# Measurement: 5 iterations, 1 s each
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Average time, time/op
# Benchmark: test.SampleBenchmark.baseLine

# Run progress: 0.00% complete, ETA 00:00:10
# Fork: 1 of 1
# Warmup Iteration   1: 1.115 ns/op
# Warmup Iteration   2: 1.122 ns/op
# Warmup Iteration   3: 1.107 ns/op
# Warmup Iteration   4: 1.108 ns/op
# Warmup Iteration   5: 1.131 ns/op
Iteration   1: 1.120 ns/op
Iteration   2: 1.321 ns/op
Iteration   3: 1.207 ns/op
Iteration   4: 1.107 ns/op
Iteration   5: 1.105 ns/op


Result: 1.172 ▒(99.9%) 0.360 ns/op [Average]
  Statistics: (min, avg, max) = (1.105, 1.172, 1.321), stdev = 0.093
  Confidence interval (99.9%): [0.812, 1.532]


# Run complete. Total time: 00:00:12

Benchmark                     Mode  Samples  Score  Score error  Units
t.SampleBenchmark.baseLine    avgt        5  1.172        0.360  ns/op
```

It seems that default include is not correct, i.e. at least it's not the same as executing JAR directly.

**Comment by marshallpierce** [comment]
Seems like all that's needed to fix this is to not have `jmh-generator-annprocess` on the compile classpath.

**Comment by quux00** [comment]
I just downloaded the jmh-gradle-plugin today and am also having this problem on Windows 7.  The workaround of doing:

    jmh {
      include = '.*SampleBenchmark.*'
    }

partially works - namely it works for benchmarks I have in `src/main/java`, but the benchmarks I have in `src/jmh/java` do not get executed even if I include them in the pattern above.  Yet those benchmark classes from `src/jmh/java` **are** in the `build/libs/my-project-SNAPSHOT-jmh.jar`.

So there are a couple of problems here to be solved and this ticket has been opened for a while.  Any chance this can get solved soon?  I'm using jmh-gradle-plugin 0.2.0, as outlined in the README.

**Comment by redhatlinux10** [comment]
i have the same problem with version 0.2.0. i also tried to set include pattern but didn't work out . i tried to delete `build` directory and this can help me out.

**Comment by npetzall** [comment]
I had a similar problem. I added jmh-profilers and it contained a BenchmarkList file which was added to the jmhJar. So two files in jmh.jar the first was the one generated by the build and the second was empty supplied by jmh-profilers which had been shadowed I suppose.

Third-party aka dependencies should have exclusions for **META-INF/BenchmarkList probably also CompilerHints.

Haven't tried the Shadow thingy if the exclude will also be applied on build output or only dependencies.

**Comment by lyrachord** [comment]
the default command line from the plugin like this
```
D:\java\JDK8\bin\java.exe  -cp C:\work\concurrent\build\libs\concurrent-jmh.jar org.openjdk.jmh.Main .* -o build\reports\jmh\human.txt -rff build\reports\jmh\results.txt
```
Faield on JDK8-b66 with jmh-1.11.2
Seems the 
```
.*
```
is the point.

**Comment by npetzall** [comment]
@lyrachord Can you verify that the concurrent-jmh.jar contains any Benchmarks with 

java -jar concurrent-jmh.jar -l

This will list all available Benchmarks. My problem was that the BenchmarkList was empty.
Also make sure that the relative paths build\reports\jmh\ exists it will fail if they don't.

**Comment by lyrachord** [comment]
@npetzall I run the whole command line, the jmh.Main failed.It works after remove the .*

**Comment by lyrachord** [comment]
@npetzall Thanks. I can list all available Benchmarks.

**Comment by sgraf812** [comment]
@lyrachord Thanks for the workaround. This works:
```
jmh {
    include=''
}
```

**Comment by vyazelenko** [comment]
Fixed with PR #53
